solver/exporter.go

```diff
@@ -136,7 +136,7 @@ func (e *exporter) ExportTo(ctx context.Context, t CacheExporterTarget, opt Cach
 				return nil, err
 			}
 			res.Release(context.TODO())
-			if remote == nil {
+			if remote == nil && len(remotes) > 0 {
```

Comment:
Shouldn't this be handled in the caller side? Why are both empty and nil needed?

Reply:
This is saying that we should only try to set `remote` to a non-nil value when there's actually anything in `remotes` to use (note that one variable is `remote` and the other is `remotes`, which is easy to miss). There are already existing similar checks for `if len(remotes) > 0` above this one on line 118, it just seemed to be missing here. `remotes` can be empty when the ref being exported is scratch I believe.